sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -0,0 +1,1107 @@
+ *     that when using import you give the appropriate permissions to the Google Cloud Healthcare
+ *     Service Agent
+ * @see <a
+ *     href=>https://cloud.google.com/healthcare/docs/how-tos/permissions-healthcare-api-gcp-products#fhir_store_cloud_storage_permissions></a>
+ * @see <a
+ *     href=>https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores/import></a>
+ *     A {@link PCollection} of {@link String} can be ingested into an Fhir store using {@link
+ *     FhirIO.Write#fhirStoresImport(String, String, String, FhirIO.Import.ContentStructure)} This
+ *     will return a {@link FhirIO.Write.Result} on which you can call {@link
+ *     FhirIO.Write.Result#getFailedBodies()} to retrieve a {@link PCollection} of {@link
+ *     HealthcareIOError} containing the {@link String} that failed to be ingested and the
+ *     exception.
+ *     <p>Example
+ *     <pre>{@code
+ * Pipeline pipeline = ...
+ *
+ * // Tail the FHIR store by retrieving resources based on Pub/Sub notifications.
+ * FhirIO.Read.Result readResult = p
+ *   .apply("Read FHIR notifications",
+ *     PubsubIO.readStrings().fromSubscription(options.getNotificationSubscription()))
+ *   .apply(FhirIO.readResources());
+ *
+ * // happily retrived messages
+ * PCollection<String> resources = readResult.getResources();
+ * // message IDs that couldn't be retrieved + error context
+ * PCollection<HealthcareIOError<String>> failedReads = readResult.getFailedReads();
+ *
+ * failedReads.apply("Write Message IDs / Stacktrace for Failed Reads to BigQuery",
+ *     BigQueryIO
+ *         .write()
+ *         .to(option.getBQFhirExecuteBundlesDeadLetterTable())
+ *         .withFormatFunction(new HealthcareIOErrorToTableRow()));
+ *
+ * output = resources.apply("Happy path transformations", ...);
+ * FhirIO.Write.Result writeResult =
+ *     output.apply("Execute FHIR Bundles", FhirIO.executeBundles(options.getFhirStore()));
+ *
+ * PCollection<HealthcareIOError<String>> failedBundles = writeResult.getFailedInsertsWithErr();
+ *
+ * failedBundles.apply("Write failed bundles to BigQuery",
+ *     BigQueryIO
+ *         .write()
+ *         .to(option.getBQFhirExecuteBundlesDeadLetterTable())
+ *         .withFormatFunction(new HealthcareIOErrorToTableRow()));
+ * }***
+ * </pre>
+ */
+public class FhirIO {
+
+  /**
+   * Read resources from a PCollection of resource IDs (e.g. when subscribing the pubsub
+   * notifications)
+   *
+   * @return the read
+   * @see Read
+   */
+  public static Read readResources() {
+    return new Read();
+  }
+
+  /**
+   * Import resources. Intended for use on empty FHIR stores
+   *
+   * @param fhirStore the fhir store
+   * @param tempDir the temp dir
+   * @param deadLetterDir the dead letter dir
+   * @param contentStructure the content structure
+   * @return the import
+   * @see Import
+   */
+  public static Import importResources(
+      String fhirStore,
+      String tempDir,
```

Comment:
You may be able to make `tempDir` optional by using the `tempLocation` pipeline option internally. You can leave this for later, as it would be a backwards-compatible change to add a new factory method. But I recommend you do do this eventually, as it's nice for users to simply forget about the temporary directories.

Reply:
Great suggestion! this should be easy to add!